chips/sam4l/src/spi.rs

```diff
@@ -305,32 +344,28 @@ impl Spi {
     pub fn set_active_peripheral(&self, peripheral: Peripheral) {
         // Slave cannot set active peripheral
         if self.role.get() == SpiRole::SpiMaster {
-            let regs: &SpiRegisters = unsafe { &*self.registers };
-            let peripheral_number: u32 = match peripheral {
-                Peripheral::Peripheral0 => spi_consts::mr::PCS0,
-                Peripheral::Peripheral1 => spi_consts::mr::PCS1,
-                Peripheral::Peripheral2 => spi_consts::mr::PCS2,
-                Peripheral::Peripheral3 => spi_consts::mr::PCS3,
+            let regs: &Registers = unsafe { &*self.registers };
+            let mr = match peripheral {
+                Peripheral::Peripheral0 => Mode::PCS::PCS0,
+                Peripheral::Peripheral1 => Mode::PCS::PCS1,
+                Peripheral::Peripheral2 => Mode::PCS::PCS2,
+                Peripheral::Peripheral3 => Mode::PCS::PCS3,
             };
-            let mut mr = regs.mr.get();
-            mr = (mr & !spi_consts::mr::PCS_MASK) | peripheral_number;
-            regs.mr.set(mr);
+            regs.mr.modify(mr);
         }
     }
 
     /// Returns the currently active peripheral
     pub fn get_active_peripheral(&self) -> Peripheral {
         if self.role.get() == SpiRole::SpiMaster {
-            let regs: &SpiRegisters = unsafe { &*self.registers };
-
-            let mr = regs.mr.get();
-            let peripheral_number = mr & (spi_consts::mr::PCS_MASK);
+            let regs: &Registers = unsafe { &*self.registers };
 
+            let peripheral_number = regs.mr.read(Mode::PCS);
             match peripheral_number {
-                spi_consts::mr::PCS0 => Peripheral::Peripheral0,
-                spi_consts::mr::PCS1 => Peripheral::Peripheral1,
-                spi_consts::mr::PCS2 => Peripheral::Peripheral2,
-                spi_consts::mr::PCS3 => Peripheral::Peripheral3,
+                0b1110 => Peripheral::Peripheral0,
+                0b1101 => Peripheral::Peripheral1,
+                0b1011 => Peripheral::Peripheral2,
+                0b0111 => Peripheral::Peripheral3,
```

Comment:
And here, this would be especially nice to replace with flags that have meaning

Reply:
Same question.